src/utils/useSong.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { Song } from '../types/song';
+import localforage, { SONGS } from './extendedLocalForage';
+
+export default function useSong(id: number | undefined) {
```

Comment:
@gary-lgy Can't handle this at callsite as it would mean the hook is conditionally rendered, which should not happen

Reply:
I see